lib/tasks/backend_api.rake

```diff
@@ -0,0 +1,8 @@
+namespace :backend_api do
+  desc 'Destroy all orphan backend apis that does not belongs to any service when api_as_product is disabled'
+  task :destroy_orphans => :environment do
+    Account.find_each do |account|
+      account.backend_apis.orphans.destroy_all unless account.provider_can_use?(:api_as_product)
```

Comment:
Or the other way around
```suggestion
BackendApi.orphans.each { |b| b.destroy unless account.provider_can_use?(:api_as_product) }
```

It will make less requests